packages/web3-providers-http/src/index.js

```diff
@@ -112,6 +112,16 @@ HttpProvider.prototype.send = function (payload, callback) {
         }
     };
 
+	//since XHR2._onHttpRequestError swallows the initial request error, we need to get it from the underlying request
+	request.addEventListener('loadstart', function() {
+		var clientRequest = request._request;
```

Comment:
If `request` is an object of `XMLHttpRequest` then I could not find any documented reference of `_request` property. We should avoid using such private behaviour of a native object. 

Also the `request` object could be either `xhr2-cookies` or `XMLHttpRequest` so such private attributes can't provide confirmation of availability on different type of objects. 


Did you tried top level events `error` or `abort`? As per docs if there is any error `error` should be fired after `loadstart`

Reply:
`_request` is a private property, and this PR would likely break if we switch out the package to use `node-xhr2`

This PR was originally created by @uluhonolulu, and I'm not really sure how to setup an environment to test that this works as intended. However, how we adjust this specific code is dependent on what package we choose to go with

Reply:
why we are binding `clientRequest.on('error'...` when request started loading `loadstart` ? shouldnt it be done directly with `request` inside `HttpProvider.prototype.send` ?

Reply:
> If `request` is an object of `XMLHttpRequest` then I could not find any documented reference of `_request` property. We should avoid using such private behaviour of a native object.
> 
> Also the `request` object could be either `xhr2-cookies` or `XMLHttpRequest` so such private attributes can't provide confirmation of availability on different type of objects.
> 
> Did you tried top level events `error` or `abort`? As per docs if there is any error `error` should be fired after `loadstart`

@nazarhussain this is my hacky attempt to get the underlying error, because xhr2-cookies just swallows it (see https://github.com/souldreamer/xhr2-cookies/issues/9).

BTW node-xhr2 has the same issue (https://github.com/pwnall/node-xhr2/blob/master/src/001-xml_http_request.coffee#L608), but at least it's developed more actively so there's a chance they fix it.

Reply:
> why we are binding `clientRequest.on('error'...` when request started loading `loadstart` ? shouldnt it be done directly with `request` inside `HttpProvider.prototype.send` ?

@jdevcs because `request._request` doesn't exist at this point yet.